src/coreclr/jit/lsra.cpp

```diff
@@ -6732,7 +6732,8 @@ void LinearScan::resolveRegisters()
 
             // Handle the DummyDefs, updating the incoming var location.
             for (; refPosIterator != refPositions.end() && currentRefPosition->refType == RefTypeDummyDef;
```

Comment:
Instead of the very long for statement and redundant checks that the compiler may or may not be able optimize out, would it be better to write it like this?
```
        for (; refPosIterator != refPositions.end(); ++refPosIterator)
        {
            currentRefPosition = &refPosIterator;
            if (currentRefPosition->refType != RefTypeParamDef && currentRefPosition->refType != RefTypeZeroInit)
                break;
```
@dotnet/jit-contrib Any preferences?

Reply:
I would personally prefer your suggestion. Also, the current jit changes have measurable throughput impact.

Reply:
The throughput impact ranges from 0.02 to 0.10% https://dev.azure.com/dnceng/public/_build/results?buildId=1924863&view=ms.vss-build-web.run-extensions-tab. Also, without enabling these sanitizer warnings in CI, it will definitely regress. We previously either suppressed such warnings or enabled them so they are visible in the CI.

Reply:
I was attempting to change as little as possible in `lsra.cpp` - some of the for loop predicates are incredibly dense. Teasing them out might have made them better. These changes need additional auditing as I am making a quick attempt on the bus.

Reply:
I think many (all?) of the `currentRefPosition` variables are not necessary.  This would partially clean up the code (the need to update both), which wouldn't directly help the conditionals but would at least clean up the area around them.

I was going to also say that the removal of the new conditionals on them might have a chance of removing the (small) throughput regression, but the new iteration here already does that.

There's no need to make that change here.  I can do it separately.  Let me know if having that in first would help you, but I'm guessing the merge conflict would do more harm than good so probably not.

Reply:
Thanks @markples. Your suggestions all make sense, a follow-up I think is most appropriate. I am done modifying this PR so as soon as it is green and I get JIT signoff I will merge it in.